solveset: Fixed NoneType Error of solveset()

RaiAlexander opened the pull request:
#### References to other Issues or PRs
Fixes #17882 


#### Brief description of what is fixed or changed
Fixes  #17882 
Fixed an error in solveset where it would return a NoneType error
Unrad() returns a None in some circumstances, so a check was added
to make sure that the function will still work even in that
circumstance. Now the original equation is passed if unrad() returns
None.

Example:

solveset(-8*x**2/(9*(x**2 - 1)**(4/3)) + 4/(3*(x**2 - 1)**(1/3)), x)
used to return NoneType Error, now returns
ConditionSet(x, Eq(-8*x**2*(x**2 - 1)**(-1.33333333333333)/9 + 4*(x**2 - 1)**(-0.333333333333333)/3, 0), S.Complexes)

#### Other comments
Added tests

#### Release Notes

<!-- Write the release notes for this release below. See
https://github.com/sympy/sympy/wiki/Writing-Release-Notes for more information
on how to write release notes. The bot will check your release notes
automatically to see if they are formatted correctly. -->

<!-- BEGIN RELEASE NOTES -->
* solvers
  * Fixed NoneType Error Bug in solveset
<!-- END RELEASE NOTES -->

RaiAlexander committed:
solveset: Fixed NoneType Error of solveset()

Fixed an error in solveset where it would return a NoneType error
Unrad() returns a None in some circumstances, so a check was added
to make sure that the function will still work even in that
circumstance. Now the original equation it passed if unrad() returns
None.
Example:
solveset(-8*x**2/(9*(x**2 - 1)**(4/3)) + 4/(3*(x**2 - 1)**(1/3)), x)
used to return NoneType Error, now returns
ConditionSet(x, Eq(-8*x**2*(x**2 - 1)**(-1.33333333333333)/9 + 4*(x**2 - 1)**(-0.333333333333333)/3, 0), S.Complexes)

jksuom commented on sympy/solvers/solveset.py:
```diff
@@ -737,7 +737,12 @@ def _has_rational_power(expr, symbol):
 def _solve_radical(f, symbol, solveset_solver):
     """ Helper function to solve equations with radicals """
-    eq, cov = unrad(f)
+    if unrad(f) is None: # Check if unrad returns NoneType
```
The result of `unrad(f)` should be saved in a local variable to avoid having to repeat the call on line 744. Maybe something like this

    res = unrad(f)
    eq, cov = res if res else (f, [])

RaiAlexander committed:
powsimp: Fixed powsimp simplify bug

Fixed bug in powsimp where the sqrt tag would be printed
out multiple times instead of one time. Now the equations
that will be printed will be shorter and easier to read

Example:

powsimp(sqrt(1-x**2)/sqrt(1-x), force=True)
outputs sqrt((1 - x**2)/(1 - x)) instead of
sqrt(1 - x**2)/sqrt(1 - x)

RaiAlexander committed:
Revert "solveset: Fixed NoneType Error of solveset()"

This reverts commit 8600b5279f6476b59dec949ab0c9aa3ff7944389.

jksuom commented:
It looks like there is some confusion. The changes are in `powsimp()`, not `solveset()` as expected.
